### DISCUSSION TITLE
Mysql updating only one table deletes all others

### DISCUSSION THREAD

**Discussion by gmentzik** [Budibase/budibase#11182] [Help] [answered]
Hi devs,

I have checked in both last weeks 2.7 and this weeks 2.8.2 version. I am facing the following serious issue!
When syncing with db to update only one table it deletes all others. This feature was supposed to be there because in previous versions update all was killing any custom table fields. Has anything changed regarding that? Is the old issue fixed and from now on we should fetch all tables?

![image](https://github.com/Budibase/budibase/assets/65217733/19b22cb4-5a25-49e7-a654-c458045e9947)

![image](https://github.com/Budibase/budibase/assets/65217733/6c2cd07a-99d8-45f2-85a9-88d78213bdbe)

![image](https://github.com/Budibase/budibase/assets/65217733/b43111e2-2eaf-4338-aec6-2aece8ad2cb7)

**Comment by luisdgeneMT** [comment]
I believe the intended behavior is:

If a table is marked in the **Fetch Tables**, that table will show up in your Tables tab.
If it is not marked, it will not show up - but it will not delete it in the database, it is just "hidden" or unsynced from Budibase.

Imagine having 300 tables in a single database and having to scroll through all of them when all you need is one table.
That is what this is trying to avoid.

If you click Fetch tables again, you should be able to sync with any or all tables once more.

**Reply by gmentzik** [reply]
Maybe i did not explain well before. I have synced my db and for example it has 10 tables. Then I modify one table in db or add a new on and select to sync just that. After that action all other tables are deleted from Budibase except the one I synced!
Fetch all is not safe as it used delete any customization done to tables in Budibase. I don't know if this was fixed. I am using version 2.5 and testing latest and fell to this issue.

Expected behavior is here:
https://docs.budibase.com/docs/sql-datasource#fetching-specific-tables
Another reason may be to prevent changes to certain tables being overwritten in Budibase; particularly [Formula](https://docs.budibase.com/docs/formula) columns.

Just to add that I tried to fetch all tables and at least for me it was inconsistent. One time it lost relationships, a second time it lost formula columns and at 3rd time it did not loose anything. So please clarify. Is the change of behavior of Fetch specific tables intentional or not?

**Reply by gmentzik** [reply]
Any news on that? Until this is clarified I cannot update to any version > 2.5.

**Reply by melohagan** [reply, marked as answer]
> Fetch all is not safe as it used delete any customization done to tables in Budibase. I don't know if this was fixed.

Hi @gmentzik this issue has been fixed. When you fetch tables it no longer removed custom formulas, or remove selected Budibase column types, etc.

The docs were outdated and have since been updated to reflect the fix. Apologies for the confusion!

**Comment by gmentzik** [comment]
Hi @melohagan.

Sorry for the delayed response. After updating to  2.8.10 I used the new backend interface to update tables and indeed formulas are not lost. There was a case where a form field also related to a db relationship was greyed out after updating table. Fixed issue by deleting and recreating relationship.
